rest/rest-cli/src/main/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/SubmitJobCommand.java

```diff
@@ -97,17 +99,17 @@ public void execute(ApplicationContext currentContext) throws CLIException {
     }
 
     private void validateFilePath(ApplicationContext currentContext) {
-        if (!isFilePathValid(pathname)) {
+
+        if (!isFilePathValid(pathname) || !isXMLFile(pathname)) {
```

Comment:
by discarding all non-xml files, you break the possibility to submit a job archive as we can see in the following code:

if (APPLICATION_XML.getMimeType().equals(contentType)) {
                 jobId = currentContext.getRestClient().submitXml(currentContext.getSessionId(),
                                                                  new FileInputStream(jobFile),
                                                                  map(this.variables));
             } else {
                 jobId = currentContext.getRestClient().submitJobArchive(currentContext.getSessionId(),
                                                                         new FileInputStream(jobFile),
                                                                         map(this.variables));
             }


You should first investigate the purpose of submitJobArchive, if it's a valid or osbolete feature.

Reply:
After inspecting the server-side code, the submitting-a-job-archive feature has been removed a while ago. The client-side part is thus obsolete. For clarity, it should be removed (can be part of this pull request or another one)